study-frontend/src/Pages/MyScreenBuilder.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { page, routes } from '../constants';
+import jwtDecode from 'jwt-decode';
+import { LoadingOutlined } from '@ant-design/icons';
+import { Spin } from 'antd';
+
+interface DecodedToken {
+  exp: number;
+  // Add other token fields here if needed
+}
+
+const antIcon = <LoadingOutlined style={{ fontSize: 24 }} spin />;
+
+export const MyScreenBuilder = () => {
+  const navigate = useNavigate  ();
+    
+
+    useEffect(() => {
+        const navigator  = () => {
```

Comment:
I will give an example here how this can be imporeved
` // Main App component
const App = () => {
  return (
  // AuthWrapper there we will validate if the user can see pages or which pages and its wrapping the router. Your validation logic  should be checked in it.
    <AuthWrapper>
      <Outlet />
    </AuthWrapper>
  );
};
`